.github/workflows/pull_request_ci.yml

```diff
@@ -0,0 +1,67 @@
+name: Pull Request CI
+
+on: [pull_request]
+
+jobs:
+  # Run code validattion
+  validation_check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - name: Setup Nix
+        uses: cachix/install-nix-action@v13
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      - name: "Install poetry"
+        run: nix-shell --run "poetry install"
+      - name: Run style check
+        run: nix-shell --run "poetry run make style_check"
+      - name: Run code check
+        run: nix-shell --run "poetry run make code_check"
+  # Run controller test
+  controller_test:
+    needs: validation_check
+    runs-on: ubuntu-latest
+    env:
+      SDL_VIDEODRIVER: "dummy"
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - name: Setup Nix
+        uses: cachix/install-nix-action@v13
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      - name: Download emulator and bridge binaries
+        run: nix-shell --run "./src/binaries/firmware/bin/download.sh && ./src/binaries/trezord-go/bin/download.sh"
+      - name: "Install poetry"
+        run: nix-shell --run "poetry install"
+      - name: Run the tests
+        run: nix-shell --run "poetry run make test"
+  # Build and push test trezor-user-env image if all validation jobs pass
+  build_images:
```

Comment:
We can rename it to `build_test_images`, so it explicitly says it is `test` and not `latest`

Reply:
And it is actually not `images`, but just one `image` - so I suggest `build_test_image` or `build_tenv_test_image`. Also it should mention `push`, so what about `build_and_push_tenv_test_image`?